src/library/tactic/list_axioms_tactic.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright (c) 2017 Robert Y. Lewis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+
+Author: Robert Y. Lewis
+*/
+#include "kernel/for_each_fn.h"
+#include "library/util.h"
+#include "library/vm/vm_name.h"
+#include "library/vm/vm_list.h"
+#include "library/tactic/tactic_state.h"
+#include "library/tactic/list_axioms_tactic.h"
+#include "library/sorry.h"
+
+namespace lean {
+
+void list_axioms_deps::visit(name const & n) {
+    if (m_visited.contains(n))
+        return;
+    m_visited.insert(n);
+    declaration const & d = m_env.get(n);
+    if (!d.is_definition() && !m_env.is_builtin(n)) {
+        m_use_axioms = true;
+        m_axioms.insert(d.get_name());
+    }
+    visit(d.get_type());
+    if (d.is_definition())
+        visit(d.get_value());
+}
+
+void list_axioms_deps::visit(expr const & e) {
+    for_each(e, [&](expr const & e, unsigned) {
+        if (is_sorry(e) && !m_used_sorry) {
+            m_used_sorry = true;
+        }
+        if (is_constant(e))
+            visit(const_name(e));
+        return true;
+    });
+}
+
+buffer<name> list_axioms_deps::axiom_list() {

```

Comment:
You can't safely copy or return buffers (since they're stack-allocated).  Just return the name_set here.